x/evm/keeper/statedb.go

```diff
@@ -630,11 +734,19 @@ func (k *Keeper) AddSlotToAccessList(addr common.Address, slot common.Hash) {
 
 // Snapshot return the index in the cached context stack
 func (k *Keeper) Snapshot() int {
+	if k.HasStateError() {
+		return -1
```

Comment:
please add a comment for this, usually the zero value should be returned

Reply:
sure

Reply:
okay, lets use `0` then